src/181810.c

```diff
@@ -108,7 +108,72 @@ ApiStatus ActorSpeak(Evt* script, s32 isInitialCall) {
     return ApiStatus_BLOCK;
 }
 
-INCLUDE_ASM(s32, "181810", EndActorSpeech);
+ApiStatus EndActorSpeech(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 flags;
+    s32 anim;
+    f32 x, y, z;
+    s32 screenX, screenY, screenZ;
+
+    if (isInitialCall) {
+        Actor* actor = (Actor*) evt_get_variable(script, *args++);
```

Comment:
This should be typed s32, get_actor(id) returns an Actor*

Reply:
He had to re-use the same var for the id and the actor itself, so it's a cast one way or the other